eth_account/_utils/typed_transactions/base.py

```diff
@@ -0,0 +1,78 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from typing import (
+    Any,
+    Dict,
+    Tuple,
+)
+
+from eth_utils import (
+    is_bytes,
+    is_string,
+    to_bytes,
+    to_int,
+)
+from eth_utils.curried import (
+    apply_formatter_to_array,
+    apply_formatters_to_dict,
+    apply_one_of_formatters,
+    hexstr_if_str,
+)
+from eth_utils.toolz import (
+    identity,
+    merge,
+)
+
+from eth_account._utils.validation import (
+    LEGACY_TRANSACTION_FORMATTERS,
+)
+
+TYPED_TRANSACTION_FORMATTERS = merge(
+    LEGACY_TRANSACTION_FORMATTERS,
+    {
+        "chainId": hexstr_if_str(to_int),
+        "type": hexstr_if_str(to_int),
+        "accessList": apply_formatter_to_array(
+            apply_formatters_to_dict(
+                {
+                    "address": apply_one_of_formatters(
+                        (
+                            (is_string, hexstr_if_str(to_bytes)),
+                            (is_bytes, identity),
+                        )
+                    ),
+                    "storageKeys": apply_formatter_to_array(hexstr_if_str(to_int)),
+                }
+            ),
+        ),
+        "maxPriorityFeePerGas": hexstr_if_str(to_int),
+        "maxFeePerGas": hexstr_if_str(to_int),
+        "maxFeePerBlobGas": hexstr_if_str(to_int),
+        "blobVersionedHashes": apply_formatter_to_array(hexstr_if_str(to_bytes)),
+    },
+)
+
+
+class _TypedTransactionImplementation(ABC):
```

Comment:
I think it makes sense to have `assert_valid_fields` there but `from_dict` and `from_bytes` return instances specific to the class that they are defined within so I think the signature differences muddy the abstract method + implementation pattern.

Reply:
I could see it going either way... `assert_valid_fields` isn't implemented at the `TypedTransaction` level so I don't think it's a necessity, more of a safety check. And it has a different signature for `BlobTransaction` that we'd have to apply across the board since we don't validate certain fields of the dictionary if blobs are also provided.

Curious on thoughts here @pacrob @kclowes

Reply:
Okay, yeah, I think it makes sense to exclude them then. Thanks for the clarification!